huaweicloud/config/config.go

```diff
@@ -89,7 +97,14 @@ func (c *Config) LoadAndValidate() error {
 		} else {
 			err = buildClientByPassword(c)
 		}
-
+	} else {
```

Comment:
should we add a switch in provider?

Reply:
I'd like to refactor this part later on with unionsdk.